pos_accounting/src/pool/delta/delta_amount_collection.rs

```diff
@@ -126,3 +98,188 @@ impl<K: Ord> Default for DeltaAmountCollection<K> {
+    }
+
+    #[test]
+    fn test_add_zero_sum() {
+        // 1 add -1 = remove
+        let mut collection = DeltaAmountCollection {
+            data: BTreeMap::from([(1, SignedAmount::from_atoms(1))]),
+        };
+
+        collection.merge_delta_amount_element(1, SignedAmount::from_atoms(-1)).unwrap();
+        assert!(collection.data.is_empty());
+    }
+
+    #[test]
+    fn test_add_overflow() {
+        let mut collection = DeltaAmountCollection {
+            data: BTreeMap::from([(1, SignedAmount::MAX)]),
+        };
+
+        let res = collection.merge_delta_amount_element(1, SignedAmount::from_atoms(1));
+        assert_eq!(res, Err(Error::ArithmeticErrorDeltaAdditionFailed));
+    }
+
+    #[test]
+    fn test_add_underflow() {
+        let mut collection = DeltaAmountCollection {
+            data: BTreeMap::from([(1, SignedAmount::MIN)]),
+        };
+
+        let res = collection.merge_delta_amount_element(1, SignedAmount::from_atoms(-1));
+        assert_eq!(res, Err(Error::ArithmeticErrorDeltaAdditionFailed));
+    }
+
+    #[test]
+    fn test_add_unsign() {
+        // 1 add 1 = 2
+        let mut collection = DeltaAmountCollection {
+            data: BTreeMap::from([(1, SignedAmount::from_atoms(1))]),
+        };
+
+        collection.add_unsigned(1, Amount::from_atoms(1)).unwrap();
+        assert_eq!(collection.data.len(), 1);
+        assert_eq!(
+            collection.data.get(&1).unwrap(),
+            &SignedAmount::from_atoms(2)
+        )
+    }
+
+    #[test]
+    fn test_add_unsign_overflow() {
+        let mut collection = DeltaAmountCollection {
+            data: BTreeMap::from([(1, SignedAmount::from_atoms(2))]),
+        };
+
+        let res = collection.add_unsigned(1, Amount::MAX);
+        assert_eq!(res, Err(Error::ArithmeticErrorToSignedFailed));
+    }
+
+    #[test]
+    fn test_sub_existing() {
+        // 2 sub 1 = 1
+        let mut collection = DeltaAmountCollection {
+            data: BTreeMap::from([(1, SignedAmount::from_atoms(2))]),
+        };
+
+        collection.sub_unsigned(1, Amount::from_atoms(1)).unwrap();
+        assert_eq!(collection.data.len(), 1);
+        assert_eq!(
+            collection.data.get(&1).unwrap(),
+            &SignedAmount::from_atoms(1)
+        )
+    }
+
+    #[test]
+    fn test_sub_zero_sum() {
+        // 1 sub 1 = remove
+        let mut collection = DeltaAmountCollection {
+            data: BTreeMap::from([(1, SignedAmount::from_atoms(1))]),
+        };
+
+        collection.sub_unsigned(1, Amount::from_atoms(1)).unwrap();
+        assert!(collection.data.is_empty());
+    }
+
+    #[test]
+    fn test_sub_unsigned_underflow() {
+        let mut collection = DeltaAmountCollection::from_data([(1, SignedAmount::from_atoms(2))]);
+
+        let res = collection.sub_unsigned(1, Amount::MAX);
+        assert_eq!(res, Err(Error::ArithmeticErrorToSignedFailed));
+    }
+
+    #[test]
+    fn test_merge_collections() {
```

Comment:
Same comment as DeltaDataCollection. This needs some heavy proprty based testing some time later.